rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -222,8 +222,15 @@ where
                 );
 
                 if let Some(tx) = self.broadcast_sender.as_ref() {
+                    // Deduplicate txids in logs to only send unique txids
+                    let mut unique_txids = HashSet::new();
                     for (_, meta) in &logs {
-                        if let Err(err) = tx.send(meta.transaction_id).await {
+                        unique_txids.insert(meta.transaction_id);
+                    }
+
+                    // Send unique txids to the receiver
```

Comment:
nit: I think the variable naming (`unique_txids`) is suggestive enough to not warrant this comment